### PR TITLE
Fix issue 14519:Parsing a GetResult returns NPE if found field is mis…

### PR DESCRIPTION
**Pull request by imvtsl**
…sing. Signed-off-by: Vatsal <vatsal.v.anand@gmail.com>

### Description
GetResult.fromXContentEmbedded() throws NullPointerException when invoking GetResult constructor if the field "found" is null. Added a null check on this field before invoking the GetResult constructor. In case, it is null, it throws a Parsing Exception with message "Missing required field [found]".

### Related Issues
Resolves #[14519]

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by imvtsl** [comment]
@dbwiddis 
Please review.

**Review by gaobinlong — APPROVED** [review]
LGTM.

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Removed
 
 ### Fixed
+- Fix NPE in GetResult if "found" field is missing ([#14552](https://github.com/opensearch-project/OpenSearch/pull/14552))
```
Comment (dbwiddis) [inline CHANGELOG.md:39]: (Nit) Looks like most of the other entries add at the end.   And I think "handle" is better than "fix" here as we're not really changing behavior, just improving exception type.

**Review by dbwiddis — APPROVED** [review]
LGTM with a very minor nit.

File: CHANGELOG.md (hunk 1 above)
Comment (imvtsl) [reply CHANGELOG.md:39]: Okay. Can i push this change in changelog to this PR now?

Comment (imvtsl) [reply CHANGELOG.md]: I updated the changelog. Moved this entry to the end and replaced "fix" with "handle"

**Review by dbwiddis — APPROVED** [review]
LGTM (again!)

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**